mne/utils/tests/test_numerics.py

```diff
@@ -421,12 +421,13 @@ def test_pca(n_components, whiten):
     X_mne = pca_mne.fit_transform(X)
     assert_array_equal(X, X_orig)
     assert_allclose(X_skl, X_mne)
+    assert pca_mne.n_components_ == pca_skl.n_components_
     for key in ('mean_', 'components_',
                 'explained_variance_', 'explained_variance_ratio_'):
         val_skl, val_mne = getattr(pca_skl, key), getattr(pca_mne, key)
         assert_allclose(val_skl, val_mne)
     if isinstance(n_components, float):
-        assert 1 < pca_mne.n_components_ < n_dim
+        assert pca_mne.n_components_ == n_dim - 1
```

Comment:
I'm surprised this is `n_dim - 1`? I would have expected `n_dim` now, with the new adjustments?

Reply:
This data is rank reduced, see a few lines above the comment:
```
X[:, -1] = np.mean(X[:, :-1], axis=-1)  # true X dim is ndim - 1
```